src/btree/bt_read.c

```diff
@@ -125,17 +126,228 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+            WT_ERR(
+              cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+            WT_ERR(__wt_update_alloc(session, &las_value, &upd, &incr, upd_type));
+            upd->txnid = las_txnid;
+            upd->durable_ts = durable_timestamp;
+            upd->start_ts = las_timestamp;
+            upd->prepare_state = prepare_state;
+        }
+
+        /* Ignore the next set of records for the same key that is already instantiated. */
+        switch (page->type) {
+        case WT_PAGE_COL_FIX:
+        case WT_PAGE_COL_VAR:
+            p = las_key.data;
+            WT_ERR(__wt_vunpack_uint(&p, 0, &recno));
+            WT_ERR(__col_instantiate(session, recno, ref, &cbt, upd));
+            break;
+        case WT_PAGE_ROW_LEAF:
+            WT_ERR(__row_instantiate(session, &las_key, ref, &cbt, upd));
+            break;
+        default:
+            WT_ERR(__wt_illegal_value(session, page->type));
+        }
+
+        /* Remove the prepared record from LAS */
+        if (!birthmark_record && upd->prepare_state == WT_PREPARE_INPROGRESS) {
+            /* Extend the buffer if needed */
+            WT_ERR(__wt_buf_extend(session, las_prepares,
+              (las_prepare_cnt + 1) * sizeof(struct las_page_prepared_updates)));
+            las_preparep = (struct las_page_prepared_updates *)las_prepares->mem + las_prepare_cnt;
+            WT_CLEAR(las_preparep->key);
+            WT_ERR(__wt_buf_set(session, &las_preparep->key, las_key.data, las_key.size));
+            las_preparep->timestamp = las_timestamp;
+            las_preparep->txnid = las_txnid;
+            las_prepare_cnt++;
+        }
+
+        if (birthmark_record) {
+            birthmarkp->instantiated = true;
+            birthmark_record = false;
+        }
+
+        upd = NULL;
+    }
+
+    __wt_readunlock(session, &cache->las_sweepwalk_lock);
+    locked = false;
+    WT_ERR_NOTFOUND_OK(ret);
+
+    for (i = 0; i < las_prepare_cnt; i++) {
+        las_preparep = (struct las_page_prepared_updates *)las_prepares->mem + i;
+        cursor->set_key(
+          cursor, las_btree_id, &las_preparep->key, las_preparep->timestamp, las_preparep->txnid);
+        cursor->remove(cursor);
+    }
+    /* Discard the cursor. */
+    WT_ERR(__wt_las_cursor_close(session, &cursor, session_flags));
+
+    /* Instantiate the remaining birthmark entries */
+    __instantiate_birthmarks(session, ref);
+    __wt_buf_free(session, &ref->page_las->max_las_key);
+    __wt_buf_free(session, &ref->page_las->min_las_key);
 
 err:
+    if (las_prepare_cnt != 0) {
+        for (i = 0, las_preparep = las_prepares->mem; i < las_prepare_cnt; i++, las_preparep++)
+            __wt_buf_free(session, &las_preparep->key);
+    }
+    __wt_scr_free(session, &las_prepares);
+    for (birthmark_cnt = 0; birthmark_cnt < ref->page_las->birthmarks_cnt; birthmark_cnt++) {
+        birthmarkp = ref->page_las->birthmarks + birthmark_cnt;
```

Comment:
Can initialize `birthmarkp` in for loop and then do `birthmarkp++`

Reply:
fixed.